src/pybind/mgr/cephadm/module.py

```diff
@@ -2536,11 +2536,14 @@ def _apply(self, spec: GenericSpec) -> str:
         return self._apply_service_spec(cast(ServiceSpec, spec))
 
     @handle_orch_error
-    def apply_tuned_profiles(self, specs: List[TunedProfileSpec]) -> str:
+    def apply_tuned_profiles(self, specs: List[TunedProfileSpec], no_overwrite: bool = False) -> str:
         outs = []
         for spec in specs:
-            self.tuned_profiles.add_profile(spec)
-            outs.append(f'Saved tuned profile {spec.profile_name}')
+            if no_overwrite and self.tuned_profiles.exists(spec.profile_name):
+                outs.append(f"Tuned profile '{spec.profile_name}' already exists (--no-overwrite is enabled)")
```

Comment:
```suggestion
                outs.append(f"Tuned profile '{spec.profile_name}' already exists (--no-overwrite flag was passed")
```
NIT, I think it's a bit clearer